crossbeam-channel/tests/golang.rs

```diff
@@ -1507,5 +1507,59 @@ mod chan {
 
 // https://github.com/golang/go/blob/master/test/ken/chan1.go
 mod chan1 {
-    // TODO
+    use super::*;
+
+    // sent messages
+    const N: usize = 1000;
+    // receiving "goroutines"
+    const M: usize = 10;
+    // channel buffering
+    const W: usize = 2;
+
+    fn r(c: Chan<usize>, m: usize, h: Arc<Mutex<[usize; N]>>) {
+        loop {
+            select! {
+                recv(c.rx()) -> rr => {
+                    if let Ok(r) = rr {
+                        let mut data = h.lock().unwrap();
+                        if data[r] != 1 {
+                            println!("r\nm={}\nr={}\nh={}\n", m, r, data[r]);
+                            panic!("fail")
+                        }
+                        data[r] = 2;
+                    }
```

Comment:
If `rr` is an error, it would be preferable for the test to fail.
```suggestion
                    let r = rr.unwrap();
                    let mut data = h.lock().unwrap();
                    if data[r] != 1 {
                        println!("r\nm={}\nr={}\nh={}\n", m, r, data[r]);
                        panic!("fail")
                    }
                    data[r] = 2;
```

Reply:
Good catch! Thanks!